Fig.Core/Settings.cs

```diff
@@ -283,40 +378,22 @@ private object Get(Type propertyType, string propertyName, string key = null, Fu
         internal void PreLoad()
         {
             _cache = new Dictionary<string, CacheEntry>(StringComparer.InvariantCultureIgnoreCase);
+
             var errors = new List<string>();
-            foreach (var propertyInfo in GetType().GetProperties())
-            {
-                try
-                {
-                    var name = propertyInfo.Name;
-                    if (name == nameof(Environment)) continue;
 
-                    var getter = propertyInfo.GetGetMethod();
-                    if (getter == null) continue;
+            var settingsType = typeof(Settings);
+            var currentType = this.GetType();
 
-                    var val = getter.Invoke(this, Array.Empty<object>());
-                    _cache[name] = new CacheEntry(val);
-                }
-                catch (TargetInvocationException tie)
-                {
-                    switch (tie.InnerException)
-                    {
-                        case KeyNotFoundException _:
-                            errors.Add(propertyInfo.Name + " not found");
-                            break;
-                        case FormatException fe:
-                            errors.Add("Can't parse " + propertyInfo.Name);
-                            break;
-                        default:
-                            errors.Add(propertyInfo.Name + " failed," + tie.InnerException?.Message);
```

Comment:
Isn't it safe to assume that TargetInvocationException always has an InnerException? In that case we can drop the ?